### DISCUSSION TITLE
SweetAlert2 loads Ukraine anthem and simultaneously blocking Vaultwarden's web interface

### DISCUSSION THREAD

**Discussion by NSGrid** [dani-garcia/vaultwarden#3918] [Q&A]
Greetings everyone! Do the developers know about the bug with the **SweetAlert2** component included in both the original Bitwarden and this alternative variant - Vaultwarden? On Bitwarden's GitHub there was even a whole incident about it. 
https://github.com/bitwarden/clients/issues/5734

Namely, after the beginning of the famous events last year, and the subsequent sanctions against Russia, a certain system component included in Bitwarden, as well as in its given alternative implementation - Vaultwarden - after some time of operation blocks its web interface in the browser and at the same time tries to download the Ukrainian anthem. **This happens if browser language is set to Russian or web vault is hosted on RU/РФ domain.**

This situation can be cured by clearing cookies and browser cache, but this is a temporary solution. After some time of using Bitwarden/Vaultwarden , all locks are restored. According to the Bitwarden GitHub incident link I provided above, it is clear that the original Bitwarden developers have identified this bug and have already taken steps to prevent this from happening again. In this regard, a big wish to the developer of Vaultwarden - please do the same.

**Comment by BlackDex** [comment]
If they have fixed it, then it will drop into Vaultwarden when we have adopted that version.

**Comment by NSGrid** [comment]
OK, in the thread about the original Bitwarden two weeks ago it was [posted](https://github.com/bitwarden/clients/issues/5734#issuecomment-1714115488) that they were finally going to remove this dependency (apparently from SweetAlert2) and were going to include all these changes in the September update. So, we'll wait, and for the changes in Vaultwarden as well. 🙏

**Comment by InvisibleRain** [comment]
It seems that SweetAlert2 has finally been removed from the September release:

```
git clone https://github.com/bitwarden/clients.git
cd clients
git checkout browser-v2023.9.1
grep -rnIi -e sweetalert -e swal
```

**Reply by NSGrid** [reply]
Great, so we're waiting for Vaultwarden docker image to be fixed.
